fdb-record-layer-core/src/main/proto/record_key_expression.proto

```diff
@@ -0,0 +1,118 @@
+/*
+ * record_key_expression.proto
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+syntax = "proto2";
+
+package com.apple.foundationdb.record;
```

Comment:
Hm, there's a bit of a decision to be made here. By leaving the package as `com.apple.foundationdb.record`, that means that this file shares a package with `record_metadata.proto` (etc). That might not be a bad thing, but I could also see us wanting to update this to something unique like `com.apple.foundationdb.record.expressions`. I don't have too strong an opinion actually, and maybe leaving it scoped to the same level as the other files is fine. It certainly decreases the diff.

Reply:
yeah I think it makes sense to use `package com.apple.foundationdb.record.expressions`, I updated the PR. It doesn't increase more diff luckily :)